src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputData) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : inputData) {
```

Comment:
A bug: You are not validating the input data before processing it. This can lead to ArrayIndexOutOfBoundsException at line 18 if the CSV file format is incorrect. Always validate your input data before processing.